src/main/java/org/reaktivity/k3po/nukleus/ext/internal/behavior/NukleusStreamFactory.java

```diff
@@ -146,7 +148,7 @@ private void onData(
             final int readableBytes = message.readableBytes();
             final OctetsFW dataExt = data.extension();
 
-            if (channel.sourceWindow() >= readableBytes)
+            if (channel.sourceWindow() >= readableBytes && data.authorization() == authorization)
```

Comment:
Should this be an `==` test or logical-and of realm (if non-zero) and logical-or of roles?

Please review all comparisons of frame authorization with required authorization.

Reply:
I want to do exact match so tests can be properly locked down.